### PR TITLE
Batch GET state API.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Batch GET state.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #107 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

File: dapr/clients/grpc/_response.py [hunk 1]
```diff
@@ -311,3 +311,93 @@ def data(self) -> bytes:
...
+        """Initializes BulkStateItem item from :obj:`runtime_v1.BulkStateItem`.
+
+        Args:
+            key (str): state's key.
+            data (Union[bytes, str]): state's data.
+            etag (str): state's etag.
+            error (str): error when state was retrieved
+
+        Raises:
+            ValueError: if the response data is not :class:`google.protobuf.any_pb2.Any`
```
Comment (mukundansundar) [inline dapr/clients/grpc/_response.py:341]: init does not raise any error. If needed type check data and raise error.

File: dapr/clients/grpc/_response.py [hunk 2]
```diff
@@ -311,3 +311,93 @@ def data(self) -> bytes:
...
+            items: Sequence[BulkStateItem],
+            headers: MetadataTuple = ()):
+        """Initializes BulkStatesResponse from :obj:`runtime_v1.GetBulkStateResponse`.
+
+        Args:
+            items (Sequence[BulkStatesItem]): the items retrieved.
+            headers (Tuple, optional): the headers from Dapr gRPC response.
+
+        Raises:
+            ValueError: if the response data is not :class:`google.protobuf.any_pb2.Any`
```
Comment (mukundansundar) [inline dapr/clients/grpc/_response.py:394]: same as above.

File: dapr/clients/grpc/client.py [hunk 3]
```diff
@@ -339,6 +341,58 @@ def get_state(
...
+            key (Sequence[str]): the keys to be retrieved
+            parallelism (int): number of items to be retrieved in parallel.
+            metadata (tuple, optional): custom metadata
+
+        Returns:
+            :class:`BulkStatesResponse` gRPC metadata returned from callee
+            and value obtained from the state store
+        """
+
+        if len(store_name) == 0 or len(store_name.strip()) == 0:
```
Comment (mukundansundar) [inline dapr/clients/grpc/client.py:375]: check if store_name is not None first

File: dapr/clients/grpc/client.py [hunk 4]
```diff
@@ -339,6 +341,58 @@ def get_state(
             data=response.data,
             headers=call.initial_metadata())
 
+    def get_states(
+            self,
+            store_name: str,
+            keys: Sequence[str],
+            parallelism: int = 1,
+            metadata: Optional[MetadataTuple] = ()) -> DaprResponse:
```
Comment (mukundansundar) [inline dapr/clients/grpc/client.py:349]: What about the metadata passed along with the call? https://github.com/dapr/python-sdk/pull/115/files#diff-1f8d2ea8f8644e8fdb58ad256d41bcc6R339 That is generally what the MetadataTuple is used for.

**Review by mukundansundar — CHANGES_REQUESTED** [review]

File: dapr/clients/grpc/client.py (hunk 4 above)
Comment (artursouza) [reply dapr/clients/grpc/client.py:349]: For some reason, it as added to bulk API request. I moved it to behave like the other calls.

File: dapr/clients/grpc/_response.py (hunk 2 above)
Comment (artursouza) [reply dapr/clients/grpc/_response.py]: removed comment.

File: dapr/clients/grpc/_response.py (hunk 1 above)
Comment (artursouza) [reply dapr/clients/grpc/_response.py]: removed comment.

**Review by mukundansundar — APPROVED** [review]

**Comment by youngbupark** [comment]
Can you please add unit-test - https://github.com/dapr/python-sdk/blob/master/tests/clients/test_dapr_grpc_client.py ?

**Comment by artursouza** [comment]
Added. Where are these tests running? Some of them are failing.

**Review by youngbupark — APPROVED** [review]
👍

**Pull request merged**